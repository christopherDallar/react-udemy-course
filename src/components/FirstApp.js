import React from 'react';
import PropTypes from 'prop-types';

// props means properties

const FirstApp = ({ regard, subtitle }) => {
  return (
    <>
      <h1>{regard}</h1>
      <p>Mi primera app</p>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  regard: PropTypes.string.isRequired,
  subtitle: PropTypes.any,
  // other: PropTypes.number.isRequired
};
FirstApp.defaultProps = {
  subtitle: '',
  // subtitle: 'Soy un subtitle',
};

export default FirstApp;
