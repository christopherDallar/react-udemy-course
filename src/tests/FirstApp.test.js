import React from 'react';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import FirstApp from './../components/FirstApp';

describe('Test on FirstApp Component', () => {
  // test('should show "Hello, I am Goku" ', () => {
  //   const regard = 'Hello, I am Goku';

  //   const { getByText } = render(
  //     <FirstApp regard={'Hello, I am Goku'} />,
  //   );

  //   // wrapper.getByText()

  //   expect(getByText(regardPruebas sobre componentes de React)).toBeInTheDocument();
  // });

  test('should show Firts App Correctly', () => {
    const regard = 'Hello, I am Goku';

    const wrapper = shallow(<FirstApp regard={regard} />);

    expect(wrapper).toMatchSnapshot(); // To Save the literal rendered component and compare with wrapper if you press 'u' update the snapshot
  });
});
