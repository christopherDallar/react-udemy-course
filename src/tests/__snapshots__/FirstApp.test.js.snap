// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on FirstApp Component should show Firts App Correctly 1`] = `
<Fragment>
  <h1>
    Hello, I am Goku
    !!!
  </h1>
  <p>
    Mi primera app
  </p>
  <p />
</Fragment>
`;
